pypesto/objective/petab_import.py

```diff
@@ -0,0 +1,363 @@
+                        and condition.preequilibrationConditionId:
+                    fixed_preequilibration_parameter_vals = condition_df.loc[
+                        # TODO: preequilibrationConditionId might not exist
+                        condition_df.conditionId == \
+                        condition.preequilibrationConditionId,
+                        fixed_parameter_ids].values
+                    edata.fixedParametersPreequilibration = \
+                        fixed_preequilibration_parameter_vals.astype(float) \
+                                                             .flatten()
+
+            # prepare measurement matrix
+            y = np.full(shape=(edata.nt(), edata.nytrue()), fill_value=np.nan)
+            # prepare sigma matrix
+            sigma_y = np.full(
+                shape=(edata.nt(), edata.nytrue()),
+                fill_value=np.nan)
+
+            # add measurements and sigmas
+            for _, measurement in cur_measurement_df.iterrows():
+                time_ix = timepoints.index(measurement.time)
+                observable_ix = observable_ids.index(
+                    f'observable_{measurement.observableId}')
+                # TODO: measurement file should contain prefix
+
+                y[time_ix, observable_ix] = measurement.measurement
+                if isinstance(measurement.noiseParameters, numbers.Number):
+                    sigma_y[time_ix, observable_ix] = \
+                        measurement.noiseParameters
+
+            # fill measurements and sigmas into edata
+            edata.setObservedData(y.flatten())
+            edata.setObservedDataStdDev(sigma_y.flatten())
+
+            # append edata to edatas list
+            edatas.append(edata)
+
+        # simulation <-> optimization parameter mapping
+        par_opt_ids = self.petab_problem.get_optimization_parameters()
+        # take sim parameter vector from model to ensure correct order
+        par_sim_ids = list(self.model.getParameterIds())
+        # par_sim_ids = self.petab_problem.get_dynamic_simulation_parameters()
+
+        parameter_mapping = \
+            petab.core.get_optimization_to_simulation_parameter_mapping(
+                condition_df=self.petab_problem.condition_df,
+                measurement_df=self.petab_problem.measurement_df,
+                parameter_df=self.petab_problem.parameter_df,
+                sbml_model=self.petab_problem.sbml_model,
+                par_opt_ids=par_opt_ids,
+                par_sim_ids=par_sim_ids
+            )
+
+        scale_mapping = \
+            petab.core.get_optimization_to_simulation_scale_mapping(
+                parameter_df=self.petab_problem.parameter_df,
+                mapping_par_opt_to_par_sim=parameter_mapping
+            )
+
+        # create objective
+        obj = AmiciObjective(
+            amici_model=self.model, amici_solver=self.solver, edata=edatas,
+            x_ids=par_opt_ids, x_names=par_opt_ids,
```

Comment:
Could try getting parameter names from petab files first and only then fall back to IDs. But also fine to leave it for now.

Reply:
ah of course. should be in parameter_df, if not empty. Yes, I'll make an issue.